Automattic/wp-calypso
# Themes: Add Builder Referral Banner

#### Changes proposed in this Pull Request

This PR adds a banner to `/themes` that link to the Upwork referral landing page.

![image](https://user-images.githubusercontent.com/4924246/50303441-e7366d80-0441-11e9-8cf0-0fb88f118a4f.png)

![image](https://user-images.githubusercontent.com/4924246/50303414-dab21500-0441-11e9-83f0-a04a0e741761.png)

#### Testing instructions

1. Ensure that you are in the test. Set variation (builderReferralBanner): In console run `localStorage.setItem( 'ABTests', '{"builderReferralThemesBanner_20181218":"builderReferralBanner"}' );`
2. Go to `/themes`.
3. Check if you see the banner above the Search bar and make sure it links to the Upwork referral landing page.
4. Make sure that dismiss button closes banner

#### Testing Tracks Events

You can turn on analytics logging by running this in the console: localStorage.setItem( 'debug', 'calypso:analytics:*' )

- Refresh after running code to active analytics logging
- On viewing the banner: `calypso:analytics:tracks Record event "calypso_upwork_themes_banner_view" called with props {site_type: "jetpack"}`
- On clicking CTA: `calypso:analytics:tracks Recording event "calypso_upwork_themes_banner_start_now_button_click" with actual props {plan: "value_bundle", site_type: "wpcom", environment: "development", environment_id: "development", site_count: 291, …}`
- On dismissing: `calypso:analytics:tracks Recording event "calypso_upwork_themes_banner_dismiss_icon_click" with actual props {plan: "value_bundle", site_type: "wpcom", environment: "development", environment_id: "development", site_count: 291, …}`

Related: https://github.com/Automattic/wp-calypso/pull/29570
